infrastructure/permission/validPermissions.go

```diff
@@ -0,0 +1,94 @@
+package permission
+
+import "github.com/ystv/web-auth/permission/permissions"
+
+// SufficientPermissionsFor takes a permission for a task and returns that permission and higher permissions that would be acceptable
+func SufficientPermissionsFor(p permissions.Permissions) (m map[string]bool) {
+	m = make(map[string]bool)
+
+	m[p.String()] = true
+
+	switch p {
+	case permissions.MenuDisabled:
+		m[permissions.MenuDisabled.String()] = true
+		return m
+	case permissions.ManageMembersAdmin:
+	case permissions.KeyCardAccess:
+	case permissions.BookingsAdmin:
+	case permissions.CalendarAdmin:
+	case permissions.CMSAdmin:
+	case permissions.Cobra:
+	case permissions.Director:
+	case permissions.EditNetStats:
+	case permissions.EmailEveryone:
+	case permissions.EquipmentAdmin:
+	case permissions.HiresAdmin:
+	case permissions.Inform:
+	case permissions.KeyListManage:
+	case permissions.MailingListAdmin:
+	case permissions.OfficerReports:
+	case permissions.Streamer:
+	case permissions.TechieTodo:
+	case permissions.VideoStats:
+	case permissions.WatchAdmin:
+		break
```

Comment:
NITPICK: You could omit these to make the function shorter.

Reply:
Fixed